### DISCUSSION TITLE
How does RabbitMQ Plugin works and its purpose.

### DISCUSSION THREAD

**Discussion by ajay-psd** [haraka/Haraka#3334] [Q&A]
Hi All

There is RabbitMQ Plugin given in Haraka,

We had an assumption that this plugin is used to queue smtp messages to Rabbitmq and when outbound is enabled all the messages is processed from rabbitMQ instead default Queue, But this is not happening.
Messages are delivering to queue and Outbound is not using rabbitMQ. 

once we checked the code and found that  there is no integration with rabbitMQ in outbound.

Is the purpose of this plugin is give developer/user option to deliver the messages to queue and write our own custom service to process those messages independently,  there is no inter-relation with Outbound with RabittMQ.

As its is not clearly mentioned the purpose in document. 
Is this assumption is correct ?, kindly suggest.

**Comment by kayrus** [comment]
I also noticed that this plugin is quite useless and I don't understand its purpose.
In my case I'm using stmp_forward and I expect that the message will be queued when the smtp_forward.host is not reachable. This actually happens, but then all these queued messaes end up in the rabbitmq forever.

Besides the queued message doesn't contain any info about the sender and recipient, therefore it is useless for further processing.

@msimerson could you please clarify this topic?

**Comment by msimerson** [comment]
I don't know, I don't use it

**Comment by kayrus** [comment]
@gauravaror maybe you can answer this question?

**Comment by joaoscotto** [comment]
I would like to know :confused:
